addons/account/models/account_bank_statement.py

```diff
@@ -809,10 +698,18 @@ def create(self, vals_list):
 
             # Otherwise field narration will be recomputed silently (at next flush) when writing on partner_id
             self.env.remove_to_compute(st_line.move_id._fields['narration'], st_line.move_id)
+
+        # No need for the user to manage their status (from Draft to Posted)
+        st_lines.move_id.action_post()
         return st_lines
 
     def write(self, vals):
         # OVERRIDE
+        if 'date' in vals:
+            # prevents error when trying to change the date of a bank statement line and the sequence of the related
+            # move does not match the date (the date is in a different period)
+            self.move_id.name = '/'
```

Comment:
You should iterate on `self` to see if the value changes otherwise you might create holes

Reply:
We need to discuss it, I need your help as the creator of the `sequence.mixin` :)
My initial idea was to erase the move and create a new one.

Reply:
I do agree with QDP , @lars-odoo ?